pkgs/by-name/co/collabora-online/package.nix

```diff
@@ -0,0 +1,121 @@
+{
+  autoreconfHook,
+  cairo,
+  cppunit,
+  fetchFromGitHub,
+  fetchNpmDeps,
+  lib,
+  libcap,
+  libpng,
+  libreoffice-collabora,
+  nodejs,
+  npmHooks,
+  pam,
+  pango,
+  pixman,
+  pkg-config,
+  poco,
+  python3,
+  rsync,
+  stdenv,
+  zstd,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "collabora-online";
+  version = "24.04.5-3";
+
+  src = fetchFromGitHub {
+    owner = "CollaboraOnline";
+    repo = "online";
+    rev = "refs/tags/cp-${finalAttrs.version}";
+    hash = "sha256-wy+vfRmKYULq2kg0MKhdxTD7HC63cTd1cYX5xN7LYgg=";
+  };
+
+  nativeBuildInputs = [
+    autoreconfHook
+    libcap
+    nodejs
+    npmHooks.npmConfigHook
+    pkg-config
+    python3
+    python3.pkgs.lxml
+    python3.pkgs.polib
+    rsync
+  ];
+
+  buildInputs = [
+    cairo
+    cppunit
+    libpng
+    pam
+    pango
+    pixman
+    poco
+    zstd
+  ];
+
+  configureFlags = [
+    "--disable-setcap"
+    "--disable-werror"
+    "--enable-silent-rules"
+    "--with-lo-path=${libreoffice-collabora}/lib/collaboraoffice"
+    "--with-lokit-path=${libreoffice-collabora.src}/include"
+  ];
+
+  env = {
+    SETCAP = "${libcap}/bin/setcap";
+  };
+
+  postPatch =
+    ''
+      cp ${./package-lock.json} ${finalAttrs.npmRoot}/package-lock.json
+      patchShebangs .
+      substituteInPlace configure.ac --replace-fail '/usr/bin/env python3' python3
+      substituteInPlace coolwsd-systemplate-setup --replace-fail /bin/pwd pwd
+    ''
+    # Use setcap'd wrappers from the wrappers dir, not from the "application path".
+    + ''
+      substituteInPlace common/JailUtil.cpp --replace-fail \
+        'Poco::Path(Util::getApplicationPath(), "coolmount").toString()' \
+        'std::string("/run/wrappers/bin/coolmount")'
+      substituteInPlace wsd/COOLWSD.cpp --replace-fail \
+        ' forKitPath += "coolforkit";' \
+        ' forKitPath = "/run/wrappers/bin/coolforkit";'
+    ''
+    # In the nix build, COOLWSD_VERSION_HASH becomes the same as COOLWSD_VERSION, e.g. `24.04.3.5`.
+    # The web server that serves files from `/browser/$COOLWSD_VERSION_HASH`, doesn't expect the
+    # hash to contain dots.
+    + ''
+      substituteInPlace wsd/FileServer.cpp --replace-fail \
+        'Poco::RegularExpression gitHashRe("/([0-9a-f]+)/");' \
+        'Poco::RegularExpression gitHashRe("/([0-9a-f.]+)/");' \
+    '';
```

Comment:
These should probably be patches.

Reply:
Honestly, at that length of the pattern (spurious hits are not too likely) I'd say either way is fine (LibreOffice packaging uses a mix, because long substitutions are easier to maintain than patches)

Reply:
Removed the first change as it's not required if `mount_namespaces` enabled. (enabled in the NixOS module by default)

Moved the second change to `./fix-file-server-regex.patch`.